lib/jnpr/junos/device.py

```diff
@@ -206,6 +206,119 @@ def port(self):
         """
         return self._port
 
+    # ------------------------------------------------------------------------
+    # property: master
+    # ------------------------------------------------------------------------
+
+    @property
+    def master(self):
+        """
+        The mastership state of the current Routing Engine.
+
+        The current Routing Engine is the RE to which the NETCONF session is
+        connected.
+
+        Note: This property is based on new-style fact gathering and the
+              value of currently cached facts. If there is a chance the
+              mastership state may have changed since the facts were cached,
+              then dev.facts_refresh() should be invoked prior to checking
+              this property. If old-style fact gathering is in use,
+              this property will return None.
+
+        :returns: True if the current RE is the master Routing Engine. False if
+                  the current RE is not the master Routing Engine. None if
+                  unable to determine the state of the current Routing Engine.
+        """
+        master = None
+
+        try:
+            # Make sure the 'current_re' fact has a value
```

Comment:
If this itself fails, then Master value will be None, Should we start by setting value to master as True in place of None?